### ISSUE TITLE
docker client server version not match

### ISSUE THREAD

**Issue by shixudongleo** [jupyter/tmpnb#148]
I follow the quick start guid, but get the following error. (I am trying in Amazon EC2 Ubuntu 14.04. )

```
[E 150805 09:12:01 dockworker:168] Encountered a Docker error with method (5 retries remain): 404 Client Error: Not Found ("client and server don't have same version (client : 1.13, server: 1.12)")
[E 150805 09:12:01 dockworker:168] Encountered a Docker error with method (4 retries remain): 404 Client Error: Not Found ("client and server don't have same version (client : 1.13, server: 1.12)")
[E 150805 09:12:01 dockworker:168] Encountered a Docker error with method (3 retries remain): 404 Client Error: Not Found ("client and server don't have same version (client : 1.13, server: 1.12)")
[E 150805 09:12:01 dockworker:168] Encountered a Docker error with method (2 retries remain): 404 Client Error: Not Found ("client and server don't have same version (client : 1.13, server: 1.12)")
[E 150805 09:12:01 dockworker:168] Encountered a Docker error with method (1 retries remain): 404 Client Error: Not Found ("client and server don't have same version (client : 1.13, server: 1.12)")
[E 150805 09:12:01 dockworker:168] Encountered a Docker error with method (0 retries remain): 404 Client Error: Not Found ("client and server don't have same version (client : 1.13, server: 1.12)")
Traceback (most recent call last):
  File "orchestrate.py", line 277, in <module>
    main()
  File "orchestrate.py", line 257, in main
    ioloop.run_sync(pool.heartbeat)
  File "/usr/local/lib/python2.7/dist-packages/tornado/ioloop.py", line 418, in run_sync
    return future_cell[0].result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/spawnpool.py", line 148, in heartbeat
    yield diagnosis.observe()
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/spawnpool.py", line 362, in observe
    "proxy": self._proxy_routes()
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 464, in callback
    result_list = [i.result() for i in children]
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/dockworker.py", line 135, in list_notebook_servers
    trunc=False)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/dockworker.py", line 171, in _with_retries
    result = yield self._with_retries(fn, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/dockworker.py", line 171, in _with_retries
    result = yield self._with_retries(fn, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/dockworker.py", line 171, in _with_retries
    result = yield self._with_retries(fn, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/dockworker.py", line 171, in _with_retries
    result = yield self._with_retries(fn, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/dockworker.py", line 171, in _with_retries
    result = yield self._with_retries(fn, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 631, in run
    yielded = self.gen.throw(*sys.exc_info())
  File "/srv/tmpnb/dockworker.py", line 174, in _with_retries
    raise e
docker.errors.APIError: 404 Client Error: Not Found ("client and server don't have same version (client : 1.13, server: 1.12)")
```

How should I fix this error?

**Comment by danielballan** [comment]
I ran into this problem and resolved it, but my notes are not great. How did you install docker?

**Comment by shixudongleo** [comment]
Actually, my first try was using apt-get to install docker.io. I checked the version of docker, it is only 1.0  both client and server. 
Then I used the installation instructions from docker docs, the problem was solved. 

In my opinion, the problem is caused from downward compatibility. The container version was built on **docker 1.3**, and my running environment is only **docker 1.0**. 

Thank you for the reply.
